scripts/main.py

```diff
@@ -87,6 +100,62 @@ def _get_image_filename(self, file_path):
             .replace(" ", "")
         )
 
+    def _deprecate_previous_if_any(
+        self, latest_issue_url: str, latest_comment_url: str
+    ):
+        """Tell the previous comment about the new one."""
+        deprecation_notice = f"__DEPRECATED__: _This screenshot is no longer up-to-date. The latest version can be found [here]({latest_comment_url})_."
+
+        url_list_comments_in_issue = (
+            f"{self.GITHUB_API_URL}/repos/{self.configuration.GITHUB_REPOSITORY}/"
+            f"issues/{self.configuration.GITHUB_PULL_REQUEST_NUMBER}/comments"
+        )
+
+        response = requests.get(
+            url_list_comments_in_issue, headers=self._request_headers
+        )
+
+        if response.status_code != 200:
+            print_message(
+                "Unable to list previous comments; we will thus not attempt to deprecate any previous comment"
+            )
+            return
+
+        elif comments := response.json():
+            # get the latest 2 comments from a bot if any
+            # FIX ME: What if there are multiple bots posting across Issues / PRs?
```

Comment:
Indeed, maybe at least check that the comment contains `self.configuration.CUSTOM_ATTACHMENT_MSG` ?

Reply:
I think it might be be easier to solve at the level of `GitHubBranchImageUploadService:AUTHOR_NAME`. I will implement the proposal.

Reply:
That looks like a complex idea, how would it work for imgur ?

Reply:
No you were actually right. Using your idea now.